src/components/dropdowns/LanguageDropdown.tsx

```diff
@@ -1,21 +1,29 @@
 import { Dropdown } from "./Dropdown";
 import i18n from "i18next";
-import { pathWithoutLanguage } from "../../routing/lang";
+import { pathWithoutLanguage, getPathLang } from "../../routing/lang";
+import { useEffect, useState } from "react";
 
-export type LanguageDropdownProps = {
-  currentPath: string;
+type LanguageDropdownProps = {
+  astroPath: string;
 };
 
-export const LanguageDropdown = ({ currentPath }: LanguageDropdownProps) => {
-  const pathWithoutLang = pathWithoutLanguage(currentPath);
+export const LanguageDropdown = ({ astroPath }: LanguageDropdownProps) => {
+  const [pathWithoutLang, setPathWithoutLang] = useState("/");
+  const [lang, setLang] = useState(getPathLang(astroPath));
 
-  return (
+  useEffect(() => {
+    const path = location.pathname;
+    setPathWithoutLang(pathWithoutLanguage(path));
+    setLang(getPathLang(path) || "");
+  }, []);
+
+  return lang ? (
     <Dropdown
-      item={i18n.language.toUpperCase()}
+      item={lang.toUpperCase()}
       options={i18n.languages.map((lang) => lang.toUpperCase())}
       onChange={(option) =>
         (location.href = `/${option.toLowerCase()}${pathWithoutLang}`)
       }
     />
-  );
+  ) : null;
```

Comment:
For some reason, I couldn't make this suggestion in one comment, but this and the previous one belong together.

```suggestion
  );
```